ruler/compat.go

```diff
@@ -0,0 +1,30 @@
+package ruler
+
+import (
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/storage/remote"
+	"golang.org/x/net/context"
+
+	"github.com/weaveworks/cortex"
+	"github.com/weaveworks/cortex/util"
+)
+
+// Pusher is an ingester server that accepts pushes.
+type Pusher interface {
+	Push(context.Context, *remote.WriteRequest) (*cortex.WriteResponse, error)
+}
+
+// appenderAdapter adapts a distributor.Distributor to prometheus.SampleAppender
```

Comment:
Btw., in case you've wondered, here's the reasoning why Prometheus's `Append()` doesn't take a context argument yet: https://github.com/prometheus/prometheus/pull/2000#discussion_r79108319

The "If there's no use for anyone right now, let's leave it out for now." leaves open the possibility of adding it in the future.